ui/plugin-system/src/runtime/DataQueriesProvider/model.ts

```diff
@@ -11,32 +11,85 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { Definition, TimeSeriesData, UnknownSpec } from '@perses-dev/core';
+import { Definition, QueryDefinition, UnknownSpec, QueryDataType } from '@perses-dev/core';
+import { UseQueryResult } from '@tanstack/react-query';
+import { useCallback, useMemo } from 'react';
+import { useListPluginMetadata } from '../plugin-registry';
 
 type QueryOptions = Record<string, unknown>;
-
-interface DataQueriesDefinitions<QueryPluginDefinition> {
-  definitions: QueryPluginDefinition[];
-}
-export interface DataQueriesProviderProps<QueryPluginDefinition = Definition<UnknownSpec>>
-  extends DataQueriesDefinitions<QueryPluginDefinition> {
+export interface DataQueriesProviderProps {
```

Comment:
`interface DataQueriesProviderProps<T = UnknownSpec>` use a generic here. That way consumers can get type safety as well.

```tsx
<DataQueriesProvider<PrometheusTimeSeriesQuery | ElasticLogsQuery> definitions={defs}>
  {children}
</DataQueriesProvider>
```